File: core/src/main/java/by/itechart/carrier/dto/CreateCarrierDto.java
package by.itechart.carrier.dto;

import by.itechart.carrier.enums.CarrierType;
import by.itechart.common.dto.AddressDto;
import lombok.Data;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

@Data
public class CreateCarrierDto {
    @NotNull
    private AddressDto address;
    @NotEmpty
    private String name;
    @NotNull
    private CarrierType carrierType;
    @NotEmpty
    private String taxNumber;
    @NotNull
    private Boolean trusted;
}

File: web-app/online-warehouse/src/app/carrier/dto/create-carrier.dto.ts
import {AddressDto} from "../../shared/address/dto/address.dto";

export class CreateCarrierDto {
  address: AddressDto;
  name: string;
  taxNumber: string;
  carrierType: string;
  trusted: boolean;


  constructor(address?: AddressDto, name?: string, taxNumber?: string, carrierType?: string, trusted?: boolean) {
    this.address = address;
    this.name = name;
    this.taxNumber = taxNumber;
    this.carrierType = carrierType;
    this.trusted = trusted;
  }
}
